crates/chainspec/src/spec.rs

```diff
@@ -358,9 +351,20 @@ impl ChainSpec {
         }
     }
 
+    fn get_or_init_sealed_genesis_header(&self) -> &SealedHeader {
+        self.genesis_header.get_or_init(|| {
+            let header = self.make_genesis_header();
+            if self.is_dev() {
+                SealedHeader::new(header, DEV_GENESIS_HASH)
+            } else {
```

Comment:
we can't do this, this shouldn't be necessary

Reply:
right, so before I impl this change, this test broke [615ca4b](https://github.com/paradigmxyz/reth/pull/14441/commits/615ca4bcfe6700acd37542589d3a0ab76b470435)
then when I implemented it, this test broke https://github.com/paradigmxyz/reth/pull/14441/commits/4228106b70b127b3918f767790426571412f9563